app/src/main/scala/org/alephium/explorer/service/BlockFlowClient.scala

```diff
@@ -117,76 +107,74 @@ object BlockFlowClient {
         endpoint: BaseEndpoint[A, B],
         uri: Uri,
         a: A
-    ): Future[Either[String, B]] =
+    ): Future[B] = {
       endpointSender
         .send(endpoint, a, uri"${uri.toString}")
-        .map(_.left.map(_.detail))
+        .flatMap {
+          case Right(res) => Future.successful(res)
+          case Left(error) =>
+            logger.error(error.detail)
+            Future.failed(NodeApiError(error.detail))
+        }
+        .recoverWith { error =>
+          Future.failed(UnreachableNode(error.getMessage))
```

Comment:
Including the exception message (`error.getMessage`) should we also keep the stacktrace of `error`? So we get failed `send` request's call-stack in the logs?

`UnreachableNode(error)`